src/main/scala/firrtl/passes/MemUtils.scala

```diff
@@ -0,0 +1,185 @@
+/*
+ Copyright (c) 2014 - 2016 The Regents of the University of
+ California (Regents). All Rights Reserved.  Redistribution and use in
+ source and binary forms, with or without modification, are permitted
+ provided that the following conditions are met:
+ * Redistributions of source code must retain the above
+ copyright notice, this list of conditions and the following
+ two paragraphs of disclaimer.
+ * Redistributions in binary form must reproduce the above
+ copyright notice, this list of conditions and the following
+ two paragraphs of disclaimer in the documentation and/or other materials
+ provided with the distribution.
+ * Neither the name of the Regents nor the names of its contributors
+ may be used to endorse or promote products derived from this
+ software without specific prior written permission.
+ IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT,
+ SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS,
+ ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF
+ REGENTS HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE. THE SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF
+ ANY, PROVIDED HEREUNDER IS PROVIDED "AS IS". REGENTS HAS NO OBLIGATION
+ TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR
+ MODIFICATIONS.
+ */
+
+package firrtl.passes
+
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.mutable.LinkedHashSet
+import scala.collection.mutable.LinkedHashMap
+import scala.collection.mutable.HashMap
+import scala.collection.mutable.ArrayBuffer
+
+import firrtl._
+import firrtl.ir._
+import firrtl.Utils._
+import firrtl.Mappers._
+import firrtl.PrimOps._
+import firrtl.Annotations._
+import firrtl.WrappedExpression._
+
+object toBits {
+  def apply(e: Expression): Expression = {
+    e match {
+      case ex: WRef => hiercat(ex,ex.tpe)
+      case ex: WSubField => hiercat(ex,ex.tpe)
+      case ex: WSubIndex => hiercat(ex,ex.tpe)
+      case t => error("Invalid operand expression for toBits!")
+    }
+  }
+  def hiercat(e: Expression, dt: Type): Expression = {
+    dt match {
+      case t:VectorType => DoPrim(PrimOps.Cat, (0 to t.size).map(i =>
+          hiercat(WSubIndex(e, i, t.tpe, UNKNOWNGENDER),t.tpe)), Seq.empty[BigInt], UIntType(UnknownWidth))
```

Comment:
This will create a DoPrim(Cat, args, ..., ...) with args.size ==t.size. This isn't correct, as a Cat operator can only have 2 arguments. You need to build a tree of concatenations.